servicetalk-http-api/src/main/java/io/servicetalk/http/api/StrategyInfluencerAwareConversions.java

```diff
@@ -25,134 +25,116 @@ private StrategyInfluencerAwareConversions() {
         requireNonNull(original);
 
-        if (original instanceof HttpExecutionStrategyInfluencer) {
-            HttpExecutionStrategyInfluencer influencer = (HttpExecutionStrategyInfluencer) original;
-            return new StrategyInfluencingStreamingConnectionFilterFactory() {
-                @Override
-                public HttpExecutionStrategy influenceStrategy(final HttpExecutionStrategy strategy) {
-                    return influencer.influenceStrategy(strategy);
-                }
-
-                @Override
-                public StreamingHttpConnectionFilter create(final FilterableStreamingHttpConnection connection) {
-                    return new ConditionalHttpConnectionFilter(predicate, original.create(connection), connection);
-                }
-            };
-        }
-        return connection -> new ConditionalHttpConnectionFilter(predicate, original.create(connection), connection);
+        return new StrategyInfluencingStreamingConnectionFilterFactory() {
+            @Override
+            public HttpExecutionStrategy influenceStrategy(final HttpExecutionStrategy strategy) {
+                return HttpExecutionStrategyInfluencer.applyInfluence(predicate,
+                        HttpExecutionStrategyInfluencer.applyInfluence(original, strategy));
+            }
+
+            @Override
+            public StreamingHttpConnectionFilter create(final FilterableStreamingHttpConnection connection) {
+                return new ConditionalHttpConnectionFilter(predicate, original.create(connection), connection);
+            }
+        };
     }
 
     static StreamingHttpClientFilterFactory toConditionalClientFilterFactory(
             final Predicate<StreamingHttpRequest> predicate, final StreamingHttpClientFilterFactory original) {
         requireNonNull(predicate);
         requireNonNull(original);
 
-        if (original instanceof HttpExecutionStrategyInfluencer) {
-            HttpExecutionStrategyInfluencer influencer = (HttpExecutionStrategyInfluencer) original;
-            return new StrategyInfluencingStreamingClientFilterFactory() {
-                @Override
-                public HttpExecutionStrategy influenceStrategy(final HttpExecutionStrategy strategy) {
-                    return influencer.influenceStrategy(strategy);
-                }
-
-                @Override
-                public StreamingHttpClientFilter create(final FilterableStreamingHttpClient client) {
-                    return new ConditionalHttpClientFilter(predicate, original.create(client), client);
-                }
-            };
-        }
-        return client -> new ConditionalHttpClientFilter(predicate, original.create(client), client);
+        return new StrategyInfluencingStreamingClientFilterFactory() {
+            @Override
+            public HttpExecutionStrategy influenceStrategy(final HttpExecutionStrategy strategy) {
+                return HttpExecutionStrategyInfluencer.applyInfluence(predicate,
```

Comment:
While it's right to include `predicate` into `HttpExecutionStrategyInfluencer` analysis, I don't see how it will match with our intention to make `HttpExecutionStrategyInfluencer` more explicit/visible. For a predicate, users usually use lambda. Should we think a bit more about the approach here?

Reply:
Agreed that it is concerning and this issue has been a good exercise. I've been thinking about what should be potentially considered an influencer since the `TimeoutHttpServiceFilter` `TimeoutFromRequest` interface. Any bit of user code that will be executed could potentially be an influencer (or potentially block). Requiring things that would naturally be functional interfaces to implement the interface seems onerous. I think that perhaps we should use specific purpose interfaces like the `TimeoutFromRequest` and provide currying utility  methods to convert between simple functional interfaces that people want to use and the special purpose interfaces. The assistance we provide could either be `default` methods providing a basic behaviour or 

```
static TimeoutFromRequest create(Function<HttpRequestMetaData, Duration> function, 
                                                             HttpExecutionStrategy requiredStrategy) {
        return new TimeoutFromRequest() {
            @Nullable
            @Override
            public Duration apply(final HttpRequestMetaData request) {
                return function.apply(request);
            }

            @Override
            public HttpExecutionStrategy influenceStrategy(final HttpExecutionStrategy strategy) {
                return strategy.merge(requiredStrategy);
            }
        };
    }
```

I want to avoid a solution which requires users to write all that tedious boiler plate to use our interfaces.